app/models/manageiq/providers/vmware/infra_manager/inventory/parser/virtual_machine.rb

```diff
@@ -71,7 +71,7 @@ def parse_virtual_machine_storage(vm_hash, props)
         cache.find(ds)&.dig(:summary, :name) == datastore_name
       end
 
-      vm_hash[:storage] = persister.storages.lazy_find(datastore._ref)
+      vm_hash[:storage] = persister.storages.lazy_find(datastore._ref) if datastore
```

Comment:
Hit this during provisioning, with streaming refresh really early on in a VM's life it doesn't have a storage yet